src/translate.py

```diff
@@ -530,7 +530,7 @@ def f64(left: "Expression", op: str, right: "Expression") -> "BinaryOp":
     def is_boolean(self) -> bool:
         return self.op in ["==", "!=", ">", "<", ">=", "<="]
 
-    def negated(self) -> "Condition":
+    def negated(self) -> Union["BinaryOp", "UnaryOp"]:
```

Comment:
I did this because there were lots of mypy errors in the callers of `.negated()` otherwise. I will try again but I'm pretty sure this was the simplest solution, and it's also in accordance with [the robustness principle](https://en.wikipedia.org/wiki/Robustness_principle) (conservative in what you send, liberal in what you accept).